src/_guides/language/effective-dart/design.md

```diff
@@ -1103,54 +1093,49 @@ or a `File` object? Is this method synchronous or asynchronous? This is clearer:
 Future<bool> install(PackageId id, String destination) => ...
 {% endprettify %}
 
-When Dart can infer the type for you, then it's fine to omit the annotation:
+In some cases, though, the type is so obvious that writing it is pointless:
 
 {:.good-style}
 <?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred)"?>
 {% prettify dart %}
 const screenWidth = 640; // Inferred as int.
 {% endprettify %}
 
-You may still wish to explicitly annotate even when inference would fill in the
-same type. If the inference relies on values or declarations from other
-libraries, you may want to annotate the type in *your* declaration so that a
-change to that other library doesn't silently change the type of your own API
-without you realizing.
+"Obvious" isn't precisely defined, but these are all good candidates:
 
-This guideline does *not* mean you can never use the dynamic type in a public
-signature, just that you should *annotate* it explicitly when you do so.
+* Literals.
+* Constructor invocations.
+* References to other constants that are explicitly typed.
+* Simple expressions on numbers and strings.
+* Factory methods like `int.parse()`, `Future.wait()`, etc. that readers are
+  expected to be familiar with.
 
+In general, err on the side of annotating. You may still wish to explicitly
+annotate even the type is obvious. If the inferred type relies on values or
```

Comment:
There is a word missing ("when"):

> You may still wish to explicitly annotate even the type is obvious.

But you might want to rewrite it anyways:

> Even when a type is obvious, you may still wish to explicitly annotate.

Reply:
Done.